dask/bag/core.py

```diff
@@ -361,6 +361,8 @@ def fold(self, binop, combine=None, initial=no_default):
         """
         a = next(names)
         b = next(names)
+        if isinstance(initial, list):
+            initial = (list2, initial)
```

Comment:
Note that we had to use the alternative `list2` function to avoid the optimizations in dask.bag that try to make everything streaming that doesn't have multiple dependents, even things marked as `(list, ...)`.